File: src/main/java/org/cgi/assesment/model/Recipes.java
package org.cgi.assesment.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

import javax.persistence.*;
import java.util.List;

@Data
@AllArgsConstructor
@Entity
@ToString
@NoArgsConstructor
@Table(name = "recipes")
public class Recipes {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    private String title;

    private String href;

    @ElementCollection
    private List<String> ingredients;


    private String thumbnail;

}

File: src/main/js/cgi-assessment-FE/src/app/core/model/recipes.ts
export class recipes{

  id: number;
  title: string;
  ingredients: string[];
  href: string;
  thumbnail: string;

  constructor(id: number, title: string, ingredients: string[], href: string, thumbnail: string ){
      this.id = id;
      this.title = title;
      this.ingredients = ingredients;
      this.href = href;
      this.thumbnail = thumbnail;
  }
}
